AppDeployToolkit/AppDeployToolkitExtensions.ps1

```diff
@@ -5701,6 +5758,115 @@ function Import-NxtIniFileWithComments {
 	}
 }
 #endregion
+#region Function Import-NxtXmlFile
+function Import-NxtXmlFile {
+	<#
+	.SYNOPSIS
+		Imports an XML file into a PowerShell object.
+	.DESCRIPTION
+		This function reads the specified XML file and converts it into a PowerShell object.
+	.PARAMETER Path
+		The path to the XML file. This parameter is mandatory.
+	.PARAMETER Encoding
+		The encoding of the XML file.
+	.PARAMETER DefaultEncoding
+		The default encoding to use if the encoding of the XML file could not be determined.
+		The best practice is to use UTF8 with BOM as the default encoding.
+	.PARAMETER ContinueOnError
+		Specifies whether the function continues to execute if an error occurs. Accepts $true or $false. Defaults to $true.
+	.OUTPUTS
+		System.Xml.XmlDocument.
+	.EXAMPLE
+		Import-NxtXmlFile -Path C:\path\to\xml\file.xml
+		This example reads the specified XML file and converts it into a PowerShell object.
+	.EXAMPLE
+		Import-NxtXmlFile -Path C:\path\to\xml\file.xml -ContinueOnError $false
+		This example reads the specified XML file and converts it into a PowerShell object. If an error occurs, the function stops executing and throws an error.
+	.LINK
+		https://neo42.de/psappdeploytoolkit
+	#>
+	[CmdletBinding()]
+	Param (
+		[Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[string]
+		$Path,
+		[Parameter(Mandatory = $false)]
+		[string]
+		[ValidateSet('Ascii', 'BigEndianUnicode', 'Default', 'Unicode', 'UTF8', 'UTF8withBom')]
+		$Encoding,
+		[Parameter(Mandatory = $false)]
+		[string]
+		[ValidateSet('Ascii', 'BigEndianUnicode', 'Default', 'Unicode', 'UTF8', 'UTF8withBom')]
+		$DefaultEncoding = 'UTF8withBom',
+		[Parameter(Mandatory = $false)]
+		[bool]
+		$ContinueOnError = $true
+	)
+	Begin {
+		## Get the name of this function and write header
+		[string]${cmdletName} = $PSCmdlet.MyInvocation.MyCommand.Name
+	}
+	Process {
```

Comment:
Should we split this function into

Import-NxtXmlFile
and
Get-NxtEncodedContent

so we dont have to redo the encoding part all the time?

Reply:
Or why dont we use builtin pwsh functions for this? This seems overly complicated for no reason